sys/net/application_layer/emcute/emcute.c

```diff
@@ -381,19 +381,33 @@ int emcute_sub(emcute_sub_t *sub, unsigned flags)
     if (gateway.port == 0) {
         return EMCUTE_NOGW;
     }
-    if (strlen(sub->topic.name) > CONFIG_EMCUTE_TOPIC_MAXLEN) {
+    if ((flags & EMCUTE_TIT_MASK) == EMCUTE_TIT_NORMAL) {
+        topic_len=strlen(sub->topic.name);
+    }
+    else if((flags & EMCUTE_TIT_MASK) == EMCUTE_TIT_PREDEF) {
+        topic_len=2;
+    }
+    else {
+        return EMCUTE_NOTSUP;
+    }
+    if (topic_len > CONFIG_EMCUTE_TOPIC_MAXLEN) {
         return EMCUTE_OVERFLOW;
     }
 
     mutex_lock(&txlock);
 
-    tbuf[0] = (strlen(sub->topic.name) + 5);
+    tbuf[0] = (uint8_t)(5 + topic_len);
     tbuf[1] = SUBSCRIBE;
-    tbuf[2] = flags;
+    tbuf[2] = flags;  
```

Comment:
```suggestion
    tbuf[2] = flags;
```

(remove the trailing whitespace)